File: backend/src/main/java/unpsjb/labprog/backend/model/Division.java
package unpsjb.labprog.backend.model;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(
    uniqueConstraints = @UniqueConstraint(columnNames = {"anio", "numDivision"})
)
@Getter
@Setter
@NoArgsConstructor
public class Division {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private int anio;
    private int numDivision;
    private String orientacion;
    
    @Enumerated(EnumType.STRING)
    private Turno turno;
}

File: frontend/cli/src/app/divisiones/division.ts
import { Turno } from '../divisiones/turno';

export interface Division {
  id: number; // Corresponde a "long" en Java
  anio: number | null; // Corresponde a "int" en Java
  numDivision: number | null; // Corresponde a "int" en Java
  orientacion: string; // Corresponde a "String" en Java
  turno: Turno; // Enum Turno
}